torch/_export/non_strict_utils.py

```diff
@@ -226,26 +232,53 @@ def make_constraints(
     if constraint_violation_error:
         raise constraint_violation_error
 
+    # create mapping from tensor id -> (dim -> user-specified constraint spec) for user tensor inputs
+    user_tensor_input_names = {
+        spec.arg.name
+        for spec in input_specs
+        if spec.kind == InputKind.USER_INPUT and isinstance(spec.arg, TensorArgument)
+    }
+    tensor_id_to_dim_constraint: Dict[int, Dict[int, Dict[str, int]]] = defaultdict(
+        defaultdict
+    )
+    for const in constraints:
+        spec = const.serializable_spec
```

Comment:
do you really need `serializable_spec` here? isn't `t_id` and `dim` already available?